Tensile/TensileCreateLibrary.py

```diff
@@ -161,26 +162,37 @@ def buildSourceCodeObjectFile(CxxCompiler, outputPath, kernelFile):
 
       compileArgs = [which('hipcc')] + hipFlags + archFlags + [kernelFile, '-c', '-o', soFilepath]
 
-      #print(' '.join(compileArgs))
       subprocess.check_call(compileArgs)
 
       coFilenames = [soFilename]
     else:
       raise RuntimeError("Unknown compiler {}".format(CxxCompiler))
 
-    extractedCOs = [os.path.join(buildPath, name) for name in coFilenames]
-    destCOs = [os.path.join(destDir, name) for name in coFilenames]
-
-    for (src, dst) in zip(extractedCOs, destCOs):
+    destCosList = []
+    if globalParameters["PackageLibrary"]:
+      for arch in archs:
+        ensurePath(os.path.join(destDir, arch))
+        archCoFilenames = [name for name in coFilenames if arch in name]
+        extractedCOs = [os.path.join(buildPath, name) for name in archCoFilenames]
+        destCOs = [os.path.join(destDir, arch, name) for name in archCoFilenames]
+        destCosList += destCOs
+        for (src, dst) in zip(extractedCOs, destCOs):
+          shutil.copyfile(src, dst)
+    else:
+      coFilenames = [name for name in coFilenames]
+      extractedCOs = [os.path.join(buildPath, name) for name in coFilenames]
+      destCOs = [os.path.join(destDir, name) for name in coFilenames]
+      destCosList += destCOs
+      for (src, dst) in zip(extractedCOs, destCOs):
         shutil.copyfile(src, dst)
 
-    return destCOs
+    return destCosList
 
 def buildSourceCodeObjectFiles(CxxCompiler, kernelFiles, outputPath):
     args = zip(itertools.repeat(CxxCompiler), itertools.repeat(outputPath), kernelFiles)
 
     coFiles = Common.ParallelMap(buildSourceCodeObjectFile, args, "Compiling source kernels",
-                                 method=lambda x: x.starmap)
+                                 method=lambda x: x.starmap, enable=False)
```

Comment:
Does this not work with parallelization?

Reply:
I added this for debugging and forgot to remove it.